### PR TITLE
LPS-179284 Add REST resource information to FDS Entry v3

### PR DESCRIPTION
**Pull request by markocikos**
### References

- [LPS-179284 Add REST context path to FDS Entry](https://issues.liferay.com/browse/LPS-179284)
- Fixes [LPS-178414 Search box returns wrong results](https://issues.liferay.com/browse/LPS-178414)
- Followup on https://github.com/liferay-frontend/liferay-portal/pull/3174
- Followup on https://github.com/liferay-frontend/liferay-portal/pull/3197

### What is the goal of this PR?

Goals of this PR:
- Preload only REST application information from DXP backend. Rely on OpenAPI for everything else.
- Add all relevant REST resource information to FDS Entry
- Scrap `entityClassName` as a unique identifier of a headless resource. It is not, that was causing LPS-178414.

Technical notes:
- Compared to v2, this is a substantial simplification. We no longer have a REST application label, we only read, display and persist REST application path. There is no awkward parsing of application select property, bundle, or special cases for Objects. @ugeortiz @4lejandrito

### What does it look like?

https://user-images.githubusercontent.com/5435511/231477718-555205ac-afdb-4b5c-864e-b634c03b043b.mp4

### Steps to reproduce

1. Add `feature.flag.LPS-164563=true` to `portal-ext.properties`. This enables FDS Views Manager.
1. Add `feature.flag.LPS-161364=true` to `portal-ext.properties`. This enables Objects relationships nested fields, used by FDS Views Manager.
1. If you previously opened Datasets page, delete `FDS Entry` and `FDS View` on Global Menu > Control Panel > Object > Objects
1. Open Global Menu > Control Panel > Object > Datasets

### CODE REVIEW

**Comment by ugeortiz** [comment]
Thanks, @markocikos, we would need to review the way we offer these REST resources and how we can categorize them in some way to offer the best possible experience.

**Comment by markocikos** [comment]
> we would need to review the way we offer these REST resources and how we can categorize them in some way to offer the best possible experience in the selection

Sounds good! I'd like only to highlight that we can move on with the simplest implementation, currently in PR. This unblocks work on FDS Fields, as we need persisted REST application path for it. Later on, we can improve selection of resources in parallel, for example by adding categories and/or human-friendly labels in origin data.

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/display/context/FDSViewsDisplayContext.java [hunk 1]
```diff
@@ -94,49 +91,21 @@ public String getFDSViewURL() {
...
-		}
-
-		return jsonArray;
-	}
-
-	private String _getName(FDSHeadlessResource fdsHeadlessResource) {
-		String name = fdsHeadlessResource.getName();
+		List<String> restApplications = _serviceTrackerList.toList();
 
-		if (name.startsWith("ObjectEntry")) {
```
Comment (dsanz) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/java/com/liferay/frontend/data/set/views/web/internal/display/context/FDSViewsDisplayContext.java:128]: Great! we got rid of this, equal treatment for all rest endpoints

File: modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/css/FDSEntries.scss [hunk 2]
```diff
@@ -1,35 +1,5 @@
...
-		max-width: 700px;
-		width: 100%;
-	}
-
-	.modal-dialog {
-		max-width: 750px;
-	}
-}
-
-.fds-entries .count {
```
Comment (dsanz) [inline modules/apps/frontend-data-set/frontend-data-set-views-web/src/main/resources/META-INF/resources/css/FDSEntries.scss:15]: I like the idea of a custom renderer for this using <ClayBadge>

**Comment by dsanz** [comment]
LGTM (after fixing the SF)

**Comment by markocikos** [comment]
ci:test:sf

**Comment by markocikos** [comment]
@mateusralv @eltonccdantas @carlosbrichete To copy from https://github.com/liferay-frontend/liferay-portal/pull/3197#issuecomment-1500110896:

> We will need to update a few FDS Manager tests with this PR, as we are changing selection that is [used in tests](https://github.com/liferay-frontend/liferay-portal/pull/3197#issuecomment-1499376409). But, before you start, I would suggest waiting for green light in reviews, as we might make additional changes.
> 
> The changes in PR will also affect testing plans for [LPS-172396 Create and Edit FDS View with basic information](https://issues.liferay.com/browse/LPS-172396) and [LPS-174141 Display the list of headless resource fields in FDS View Fields tab](https://issues.liferay.com/browse/LPS-174141)

This PR is now finalized. Please update tests, or at least quarantine them, an update later on.

**Comment by mateusralv** [comment]
Hi @markocikos I will quarantine them, to fix them later. :)

**Comment by mateusralv** [comment]
Hi @markocikos Instead of quarantining the tests I fix all of them to the new implementations.

**Comment by markocikos** [comment]
ci:forward

**Comment by mateusralv** [comment]
@markocikos Two tests continued to fail in the CI, so I fix them and updated the PR.

**Comment by mateusralv** [comment]
ci:forward

**Comment by mateusralv** [comment]
ci:test:sf

**Comment by mateusralv** [comment]
ci:forward

**Comment by markocikos** [comment]
ci:forward

**Comment by mateusralv** [comment]
ci:forward

**Comment by mateusralv** [comment]
ci:forward

**Pull request closed without merging**